lib/ansible/modules/cloud/amazon/iam_cert.py

```diff
@@ -225,14 +249,28 @@ def cert_action(module, iam, name, cpath, new_name, new_path, state,
             module.exit_json(changed=changed, msg='Certificate with the name %s already absent' % name)
 
 
+def read_files(cert, cert_body, key, key_body, cert_chain, cert_chain_body):
+    # if paths are provided rather than lookups read the files and return the contents
+    if not cert_body and os.path.exists(cert):
```

Comment:
As previously discussed on https://github.com/ansible/ansible/pull/20787 I much prefer os.path.isfile
That way it won't try to read a directory path.